pkg/services/sso/keycloak_service_builder.go

```diff
@@ -8,34 +8,103 @@ import (
 
 var _ KeycloakServiceBuilderSelector = &keycloakServiceBuilderSelector{}
 var _ KeycloakServiceBuilder = &keycloakServiceBuilder{}
+var _ KFMKeycloakServiceBuilderConfigurator = &keycloakBuilderConfigurator{}
+var _ OSDKeycloakServiceBuilderConfigurator = &osdBuilderConfigurator{}
+
+type KeycloakServiceBuilderSelector interface {
+	ForOSD() OSDKeycloakServiceBuilderConfigurator
+	ForKFM() KFMKeycloakServiceBuilderConfigurator
+}
+
+type KFMKeycloakServiceBuilderConfigurator interface {
+	WithConfiguration(config *keycloak.KeycloakConfig) KeycloakServiceBuilder
+}
+
+type OSDKeycloakServiceBuilderConfigurator interface {
+	WithConfiguration(config *keycloak.KeycloakConfig) OSDKeycloakServiceBuilder
+}
 
 type KeycloakServiceBuilder interface {
 	WithRealmConfig(realmConfig *keycloak.KeycloakRealmConfig) KeycloakServiceBuilder
 	Build() KeycloakService
 }
 
+type OSDKeycloakServiceBuilder interface {
+	WithRealmConfig(realmConfig *keycloak.KeycloakRealmConfig) OSDKeycloakServiceBuilder
+	Build() OSDKeycloakService
+}
+
+type keycloakServiceBuilderSelector struct {
+}
+
+func (s *keycloakServiceBuilderSelector) ForOSD() OSDKeycloakServiceBuilderConfigurator {
+	return &osdBuilderConfigurator{}
+}
+
+func (s *keycloakServiceBuilderSelector) ForKFM() KFMKeycloakServiceBuilderConfigurator {
+	return &keycloakBuilderConfigurator{}
+}
```

Comment:
Why the `.ForKFM()` name? How is this method meant to be used by other fleet managers? KFM and connectors as well? 
This file is in the core service and with the name `.ForKFM()` it seems like it is specific for kas-fleet-manager, is this true?

@ziccardi @akoserwal